pkg/trait/route_test.go

```diff
@@ -538,3 +540,55 @@ func TestRoute_WithCustomServicePort(t *testing.T) {
 		route.Spec.Port.TargetPort.StrVal,
 	)
 }
+
+func TestRouteAnnotation(t *testing.T) {
+	annotationsTest := map[string]string{"haproxy.router.openshift.io/balance": "true"}
+	name := xid.New().String()
+	environment := createTestRouteEnvironment(t, name)
+	traitsCatalog := environment.Catalog
+	err := traitsCatalog.apply(environment)
+	assert.Nil(t, err)
+
+	route := environment.Resources.GetRoute(func(r *routev1.Route) bool {
+		return r.ObjectMeta.Name == name
+	})
+
+	assert.NotNil(t, route)
+	assert.True(t, reflect.DeepEqual(route.ObjectMeta.Annotations, map[string]string{}))
+
+	route = createTestRoute(environment)
+	environment.Resources.Add(route)
+
+	assert.NotNil(t, environment.GetTrait("route"))
+
+	assert.NotNil(t, route)
+	assert.True(t, reflect.DeepEqual(route.ObjectMeta.Annotations, annotationsTest))
+
+}
+
+func createTestRoute(e *Environment) *routev1.Route {
```

Comment:
This one is probably not needed. It's the logic of the operator that has to create a Route object as explained in the previous comment.

Reply:
Yes I agree, this one is not necessary.